src/_guides/language/effective-dart/design.md

```diff
@@ -974,79 +1060,260 @@ var names = people.map((Person person) {
 });
 {% endprettify %}
 
+In rare cases, the surrounding context is not precise enough to provide a type
+for one or more of the function's parameters. In those cases, you may need to
+annotate.
+
 
-### AVOID annotating with `dynamic` when not required.
+### PREFER annotating with `dynamic` even when not required.
 
-{% comment %}
-update-for-dart-2.0
-https://github.com/dart-lang/site-www/pull/480#discussion_r157355593
-{% endcomment %}
+When you omit a type annotation, *most* of the time, Dart infers a correct
+static type for you. That word "most" is the alarming part. Sometimes, inference
+fails and silently uses `dynamic` instead. If you didn't realize that inference
```

Comment:
I'm hoping that we can avoid the use of "fails" here. I'm not sure how to word it, but maybe using terminology like "defaults to `dynamic`"? (Will have to think about this more.)

Reply:
As above, I do think "fails" is an OK word.